hal/include/HAL/Task.h

```diff
@@ -12,7 +12,7 @@
 
 #ifndef _STATUS_DEFINED
 #define _STATUS_DEFINED
-typedef int STATUS;
+typedef int32_t STATUS;
```

Comment:
@ThadHouse: Not related to this PR, but shouldn't we be prefixing TASK and STATUS with HAL_ to avoid namespace pollution?